new-device-notification.php

```diff
@@ -62,7 +63,10 @@ public function start() {
 		$this->set_cookie();
 
 		// Maybe we've seen this user+IP+agent before but they don't accept cookies?
-		$memcached_key = 'lastseen_' . $current_user->ID . '_' . md5( $_SERVER['REMOTE_ADDR'] . '|' . $_SERVER['HTTP_USER_AGENT'] );
+		// phpcs:ignore WordPressVIPMinimum.Variables.ServerVariables.UserControlledHeaders
+		$remote_addr = isset( $_SERVER['REMOTE_ADDR'] ) ? $_SERVER['REMOTE_ADDR'] : '';
+		$user_agent  = isset( $_SERVER['HTTP_USER_AGENT'] ) ? $_SERVER['HTTP_USER_AGENT'] : '';
```

Comment:
Seems like these could be retrieved via `filter_input()`, and then sanitized as a text field, instead of just adding an ignore?